### ISSUE TITLE
jasper_dead_letters exception_message field is too narrow causing SQL truncation error

### ISSUE THREAD

**Issue by TheFellow** [JasperFx/jasper#669]
The error messages reported from EF Core can be quite long and when they exceed the **250** character limit it results in a failure to move to the dead-letter queue.

An anonymized version of the exception_message reported by EF Core when I broke a NULL constraint (has length **859**):
```
Cannot insert the value NULL into column 'ABC_Id', table 'App_Db.dbo.ModuleEntity'; column does not allow nulls. INSERT fails.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_ModuleProcessing_ModuleEntity".
The conflict occurred in database "App_Db", table "dbo.ModuleEntity", column 'DEF_Id'.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_ModuleProcessingStatus_ModuleProcessing".
The conflict occurred in database "App_Db", table "dbo.ModuleProcessing", column 'GHI_Id'.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_ModuleEntityDocument_ModuleEntity".
The conflict occurred in database "App_Db", table "dbo.ModuleEntity", column 'DEF_Id'.
 The statement has been terminated.
 The statement has been terminated.
 The statement has been terminated.
 The statement has been terminated.
```

There is Polly policy retrying the insert and it fails every time with `String or binary data would be truncated`

Stack trace:
```
[09:44:00 ERR] Exception detected: [] {"SourceContext": "Jasper.Transports"}
System.Data.SqlClient.SqlException (0x80131904): String or binary data would be truncated.
The statement has been terminated.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString)
   at System.Data.SqlClient.SqlCommand.CompleteAsyncExecuteReader()
   at System.Data.SqlClient.SqlCommand.EndExecuteNonQueryInternal(IAsyncResult asyncResult)
   at System.Data.SqlClient.SqlCommand.EndExecuteNonQuery(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location ---
   at Jasper.Persistence.SqlServer.Persistence.SqlServerEnvelopePersistence.MoveToDeadLetterStorage(ErrorReport[] errors)
   at Polly.AsyncPolicy.<>c__DisplayClass40_0.<<ImplementationAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Polly.Retry.AsyncRetryEngine.ImplementationAsync[TResult](Func`3 action, Context context, CancellationToken cancellationToken, ExceptionPredicates shouldRetryExceptionPredicates, ResultPredicates`1 shouldRetryResultPredicates, Func`5 onRetryAsync, Int32 permittedRetryCount, IEnumerable`1 sleepDurationsEnumerable, Func`4 sleepDurationProvider, Boolean continueOnCapturedContext)
ClientConnectionId:38e7aaca-157f-4e07-8e26-dae9db471ec3
```

**Comment by jeremydmiller** [comment]
@TheFellow OMG, I'm so sorry for not seeing all your issues. I'll be switching back to Jasper development in the next couple weeks and aiming for a 2.0 release by the end of spring.

**Comment by TheFellow** [comment]
@jeremydmiller That's great news! I had cloned the repo to try and see if I could fix a few items myself but hadn't got there.

There's a more severe one occurring that may be related to #665 where `IDisposable` instances aren't correctly disposed.
I'm using Serilog to log items being processed (built on top of Jasper obviously) and when I switched it to logging to SQL (crucially, a batching logger) I'm dropping tail-end log entries. I'll setup a minimal repro and open an issue with it soon. It's coming up at work and dropping the entries in this scenario is a no-go.

Thank you (and all contributors) for Jasper! It's exactly what I needed :dart: for this project, and once I can button up some of these items it'll be out in production soon!
